Documentation/gettingstarted_kafka.rst

```diff
@@ -0,0 +1,392 @@
+
+Obviously, it would be much more secure to limit each pod's access to the Kafka broker to be
+least privilege (i.e., only what is needed for the app to operate correctly and nothing more).
+
+We can do that with the following Cilium security policy.   As with Cilium HTTP policies, we can write
+policies that identify pods by labels, and then limit the traffic in/out of this pod.  In
+this case, we'll create a policy that identifies exactly the traffic that should be allowed to reach the
+Kafka broker, and deny the rest.
+
+As an example, a policy could limit containers with label *app=empire-outpost* to only be able to consume
+topic *empire-announce*, but would block any attempt by a compromised container (e.g., empire-outpost-9999)
+from producing to *empire-announce* or consuming from *deathstar-plans*.
+
+.. image:: images/cilium_kafka_gsg_attack.png
+
+Here is the CiliumNetworkPolicy rule that limits access of pods with label *app=empire-outpost* to
+only consume on topic *empire-announce*:
+
+::
+
+  apiVersion: "cilium.io/v2"
+  kind: CiliumNetworkPolicy
+  description: "enable outposts to consume empire-announce"
+  metadata:
+    name: "rule2"
+  spec:
+    endpointSelector:
+      matchLabels:
+        app: kafka
+    ingress:
+    - fromEndpoints:
+      - matchLabels:
+          app: empire-outpost
+      toPorts:
+      - ports:
+        - port: "9092"
+          protocol: TCP
+        rules:
+          kafka:
+          - apiKey: "fetch"
+            topic: "empire-announce"
+          - apiKey: "apiversions"
+          - apiKey: "metadata"
+          - apiKey: "findcoordinator"
+          - apiKey: "joingroup"
+          - apiKey: "leavegroup"
+          - apiKey: "syncgroup"
+          - apiKey: "offsets"
+          - apiKey: "offsetcommit"
+          - apiKey: "offsetfetch"
+          - apiKey: "heartbeat"
+
+
+CiliumNetworkPolicies operate by white-listing allowed traffic, meaning that requests that do not match
+any rules are denied as invalid.
+
+The above rule applies to inbound (i.e., "ingress") connections to kafka-broker pods (as
+indicated by "app: kafka"
+in the "endpointSelector" section).  The rule will apply to connections from pods with label
+"app: empire-outpost" as indicated by the "fromEndpoints" section.   The rule explicitly matches
+Kafka connections destined to TCP 9092, and white-lists a list of protocol requests according to various
+fields found in Kafka protocol messages.  Specifically, each request contains an "apiKey" field that
+indicates the type of request, and we white-list several apiKey's here, the most important being the
+apiKey "fetch",
+which allows the client to consume from a particular topic.   Notice that requests with "apiKey: fetch"
+are further limited by a "topic" field, in this case "empire-announce".
+
+The full policy adds two additional rules that permit the legitimate produce request
+(topic "empire-announce" and topic "deathstar-plans") from *empire-hq* and the
```

Comment:
Choose consistent pattern for 3 topic examples in this sentence: (topic "XXX") or (topic = "XXX").